test/cypress/features/common/common.js

```diff
@@ -47,50 +49,106 @@ Given(/^I am on (.*?) page$/, function (page) {
       break;
     case 'delegates':
       cy.visit(urls.delegates);
-      cy.get(ss.delegateName);
+      cy.route('/api/delegates**').as('requestDelegate');
+      cy.wait('@requestDelegate');
+      break;
+    case 'wallet':
+      cy.route('/api/transactions?*').as('transactions');
+      cy.route('/api/votes?*').as('votes');
+      cy.visit(urls.wallet);
+      cy.wait('@transactions');
+      cy.wait('@votes');
       break;
     default:
       cy.visit(urls[page]);
       break;
   }
 });
 
-Then(/^The latest transaction is (.*?)$/, function (transactionType) {
-  if (transactionType.indexOf('transfer') > 0) {
-    const transactionAddress = transactionType.split(' ').pop(); // For uses like: 'transfer to 123456L'
-    cy.get(`${ss.transactionRow} ${ss.transactionAddress}`).eq(0).should('have.text', transactionAddress);
-  } else {
-    cy.get(`${ss.transactionRow} ${ss.transactionAddress}`).eq(0).contains(/delegate vote/i);
+Given(/^I am on (.*?) page of (.*?)$/, function (page, identifier) {
+  cy.server();
+  switch (page.toLowerCase()) {
+    case 'wallet':
+      cy.route('/api/transactions?*').as('transactions');
+      cy.visit(`${urls.accounts}/${accounts[identifier].address}`);
+      cy.wait('@transactions');
+      cy.wait('@transactions');
+      cy.wait('@transactions');
+      cy.wait('@transactions');
+      cy.wait('@transactions');
+      break;
   }
 });
 
-Then(/^I should be on Tx Details page$/, function () {
-  cy.get(ss.app).contains('Confirmations');
-});
-
-Then(/^I should be on Account page$/, function () {
-  cy.get(ss.accountName).should('be.visible');
+Then(/^The latest transaction is (.*?)$/, function (transactionType) {
+  cy.wait(1000);
+  if (transactionType.toLowerCase().indexOf('transfer') > -1) {
+    cy.get(`${ss.transactionRow} ${ss.spinner}`).should('be.visible');
+    const transactionRecipient = transactionType.split(' ').pop();
+    if (transactionRecipient === 'random') {
+      /* For use: 'transfer to random' */
+      cy.get(`${ss.transactionRow} ${ss.transactionAddress}`).eq(0).should('have.text', this.randomAddress);
+    } else if (transactionRecipient === 'transfer') {
+      throw new Error('Usage: "transfer to {recipient}" where recipient could be "random" or account name');
+    } else {
+      /* For uses like: 'transfer to genesis' */
+      cy.get(`${ss.transactionRow} ${ss.transactionAddress}`).eq(0).should('have.text', accounts[transactionRecipient]);
+    }
+    cy.get(`${ss.transactionRow} ${ss.spinner}`, { timeout: txConfirmationTimeout }).should('be.not.visible');
+  }
+  switch (transactionType.toLowerCase()) {
+    case 'delegate vote':
+    case 'voting':
+      cy.get(`${ss.transactionRow} ${ss.transactionAddress}`).eq(0).contains('Delegate vote');
+      break;
+    case 'second passphrase registration':
+    case 'passphrase registration':
```

Comment:
IMO it would be more consistent to use just one of `'second passphrase registration'` and `'passphrase registration'` throughout the cypress tests.

Reply:
yeah, lets use second passphrase registration